llvm/test/Transforms/InstCombine/phi-with-multiple-unsimplifiable-values.ll

```diff
@@ -0,0 +1,105 @@
+; CHECK:       [[EXIT]]:
+; CHECK-NEXT:    [[R:%.*]] = phi i1 [ [[TMP0]], %[[TRUE]] ], [ [[TMP1]], %[[FALSE]] ]
+; CHECK-NEXT:    ret i1 [[R]]
+;
+{
+entry:
+  br i1 %c, label %true, label %false
+true:
+  %cmp1 = call i8 @llvm.scmp(i16 %x, i16 %y)
+  br label %exit
+false:
+  %cmp2 = call i8 @llvm.scmp(i16 %y, i16 %x)
+  br label %exit
+exit:
+  %phi = phi i8 [%cmp1, %true], [%cmp2, %false]
+  %r = icmp slt i8 %phi, 0
+  ret i1 %r
+}
+
+; Negative test: the RHS of comparison that uses the phi node is not constant
+define i1 @icmp_of_phi_of_scmp_with_non_constant(i1 %c, i16 %x, i16 %y, i8 %cmp)
+; CHECK-LABEL: define i1 @icmp_of_phi_of_scmp_with_non_constant(
+; CHECK-SAME: i1 [[C:%.*]], i16 [[X:%.*]], i16 [[Y:%.*]], i8 [[CMP:%.*]]) {
+; CHECK-NEXT:  [[ENTRY:.*:]]
+; CHECK-NEXT:    br i1 [[C]], label %[[TRUE:.*]], label %[[FALSE:.*]]
+; CHECK:       [[TRUE]]:
+; CHECK-NEXT:    [[CMP1:%.*]] = call i8 @llvm.scmp.i8.i16(i16 [[X]], i16 [[Y]])
+; CHECK-NEXT:    br label %[[EXIT:.*]]
+; CHECK:       [[FALSE]]:
+; CHECK-NEXT:    [[CMP2:%.*]] = call i8 @llvm.scmp.i8.i16(i16 [[Y]], i16 [[X]])
+; CHECK-NEXT:    br label %[[EXIT]]
+; CHECK:       [[EXIT]]:
+; CHECK-NEXT:    [[PHI:%.*]] = phi i8 [ [[CMP1]], %[[TRUE]] ], [ [[CMP2]], %[[FALSE]] ]
+; CHECK-NEXT:    [[R:%.*]] = icmp slt i8 [[PHI]], [[CMP]]
+; CHECK-NEXT:    ret i1 [[R]]
+;
+{
+entry:
+  br i1 %c, label %true, label %false
+true:
+  %cmp1 = call i8 @llvm.scmp(i16 %x, i16 %y)
+  br label %exit
+false:
+  %cmp2 = call i8 @llvm.scmp(i16 %y, i16 %x)
+  br label %exit
+exit:
+  %phi = phi i8 [%cmp1, %true], [%cmp2, %false]
+  %r = icmp slt i8 %phi, %cmp
+  ret i1 %r
+}
+
+; Negative test: more than one incoming value of the phi node is not one-use
+define i1 @icmp_of_phi_of_scmp_with_constant_not_one_use(i1 %c, i16 %x, i16 %y)
+; CHECK-LABEL: define i1 @icmp_of_phi_of_scmp_with_constant_not_one_use(
+; CHECK-SAME: i1 [[C:%.*]], i16 [[X:%.*]], i16 [[Y:%.*]]) {
+; CHECK-NEXT:  [[ENTRY:.*:]]
+; CHECK-NEXT:    br i1 [[C]], label %[[TRUE:.*]], label %[[FALSE:.*]]
+; CHECK:       [[TRUE]]:
+; CHECK-NEXT:    [[CMP1:%.*]] = call i8 @llvm.scmp.i8.i16(i16 [[X]], i16 [[Y]])
+; CHECK-NEXT:    call void @use(i8 [[CMP1]])
+; CHECK-NEXT:    br label %[[EXIT:.*]]
+; CHECK:       [[FALSE]]:
+; CHECK-NEXT:    [[CMP2:%.*]] = call i8 @llvm.scmp.i8.i16(i16 [[Y]], i16 [[X]])
+; CHECK-NEXT:    call void @use(i8 [[CMP2]])
+; CHECK-NEXT:    br label %[[EXIT]]
+; CHECK:       [[EXIT]]:
+; CHECK-NEXT:    [[PHI:%.*]] = phi i8 [ [[CMP1]], %[[TRUE]] ], [ [[CMP2]], %[[FALSE]] ]
+; CHECK-NEXT:    [[R:%.*]] = icmp slt i8 [[PHI]], 0
+; CHECK-NEXT:    ret i1 [[R]]
+;
+{
+entry:
+  br i1 %c, label %true, label %false
+true:
+  %cmp1 = call i8 @llvm.scmp(i16 %x, i16 %y)
+  call void @use(i8 %cmp1)
+  br label %exit
+false:
+  %cmp2 = call i8 @llvm.scmp(i16 %y, i16 %x)
+  call void @use(i8 %cmp2)
+  br label %exit
+exit:
+  %phi = phi i8 [%cmp1, %true], [%cmp2, %false]
+  %r = icmp slt i8 %phi, 0
+  ret i1 %r
```

Comment:
Any tests where only 1 of the incoming values is an `{s,u}cmp` intrin?

Reply:
Added one